migration/migrator_validate.go

```diff
@@ -138,3 +140,138 @@ func (m *StateMigrator) ValidateMigratedState(mptRoot common.Hash, zkRoot common
 
 	return nil
 }
+
+func (m *StateMigrator) ValidateNewState(num uint64, mptRoot common.Hash, stateChanges *core.StateChanges) error {
+	if num == 0 {
+		return fmt.Errorf("block number must bigger than 0")
+	}
+
+	parent := m.backend.BlockChain().GetHeaderByNumber(num - 1)
+	if parent == nil {
+		return fmt.Errorf("failed to get parent header by number %d", num-1)
+	}
+	header := m.backend.BlockChain().GetHeaderByNumber(num)
+	if header == nil {
+		return fmt.Errorf("failed to get header by number %d", num)
+	}
+	zkt, err := trie.NewZkMerkleStateTrie(parent.Root, m.zktdb)
+	if err != nil {
+		return fmt.Errorf("failed to create zk state trie: %w", err)
+	}
+	zkt.WithTransformKey(func(key []byte) ([]byte, error) {
+		secureKey, err := zktrie.ToSecureKey(key)
```

Comment:
Don't we need this part?
```suggestion
		sanityCheckByte32Key(key)
		secureKey, err := zktrie.ToSecureKey(key)
```

Reply:
I think we can ensure that keys are limited to slots or addresses during validation. You can refer to lines 178, 185–186, and 194–195 below.